java/arcs/sdk/Particle.kt

```diff
@@ -66,10 +65,12 @@ open class HandleHolderBase(
         handles[handleName] = handle
     }
 
+    override fun detach() {
+        handles.forEach { (_, handle) -> handle.unregisterForStorageEvents() }
+    }
+
     override fun reset() {
-        runBlocking {
-            handles.forEach { (_, handle) -> handle.close() }
-        }
+        detach()
```

Comment:
It's unclear to me why reset() wouldn't close()?

Reply:
Ah, yeah, this was an "overdelete", I was just trying to remove the `runBlocking` and removed the whole thing.

I added the `detach` here so that if `close` is called without calling `detach` beforehand, things still get cleaned up.